File: 0070-climbing-stairs/0070-climbing-stairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        
        if (n<4) {
            return n;
        }
        int a = 2,b = 3,c,j=0;
        while(j<n-3) {
            c = a+b;
            a = b;
            b = c;
            j++;
        }
        
        return c;
        
    }
};

File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        '''
        cache = {}
        
        def helper(n):
            
            if n in cache:
                return cache[n]
            
        
    
            if n <3:
                res =  n
            
            else:
                res = helper(n-1) + helper(n-2)
                
            
        
            cache[n] = res
            return res
        return helper(n)
        ''' 
        if n<3:
            return n
        dp = [0]*n
        dp[0] = 1
        dp[1] = 2
        for i in range(2,n):
            dp[i] = dp[i-1] + dp[i-2]
        return dp[n-1]